src/Unity.Mathematics/random.cs

```diff
@@ -27,6 +27,45 @@ public Random(uint seed)
             NextState();
         }
 
+        /// <summary>
+        /// Constructs a <see cref="Random"/> instance with a hashed seed.
+        /// </summary>
+        /// <remarks>
+        /// Use this function when you expect to create several Random instances from consecutive or
+        /// very similar seeds.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// for (uint i = 0; i &lt; 4096; ++i)
+        /// {
+        ///     Random rand = Random.CreateFromHashedSeed(i);
+        ///
+        ///     // Random numbers drawn from loop iteration j should be very different
+        ///     // from every other loop iteration.
+        ///     rand.NextUInt();
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="seed_to_hash">A seed that will be hashed for Random creation.</param>
+        /// <param name="backup_nonzero_seed">In the event that <paramref name="seed_to_hash"/> hashes to zero, then the <see cref="Random"/> constructor will be called with this backup value as the seed instead.  No hashing is performed on the backup and it must be non-zero.</param>
+        /// <returns><see cref="Random"/> created from hashed seed.  If the seed hashes to zero, then the backup seed is used without hashing it.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Random CreateFromHashedSeed(uint seed_to_hash, uint backup_nonzero_seed = 1u)
+        {
+            // Wang hash: this has the property that none of the outputs will
```

Comment:
Provide a reference in the comments?

Reply:
The original site has been down for some time but someone backed it up https://gist.github.com/badboy/6267743

The relevant section is https://gist.github.com/badboy/6267743#hash-function-construction-principles.  I'll add it to the comment.